File: 2732/main.go
package main

import "fmt"

func main() {
	fmt.Println(goodSubsetofBinaryMatrix([][]int{
		{0,1,1,0},
		{0,0,0,1},
		{1,1,1,1},
	}))
}

func goodSubsetofBinaryMatrix(grid [][]int) []int {
	ans := []int{}
	mp := make(map[int]int)
	m := len(grid)
	n := len(grid[0])
	for j := 0; j < m; j++ {
		st := 0
		for i := 0; i < n; i++ {
			st |= (grid[j][i] << i)
		}
		mp[st] = j
	}
	if _, ok := mp[0]; ok {
		ans = append(ans, mp[0])
		return ans
	}
	for x, i := range mp {
		for y, j := range mp {
			if x & y == 0 {
				return []int{min(i, j), max(i, j)}
			}
		}
	}
	return ans
}

File: 2732/main.rs
use std::collections::HashMap;
struct Solution{}

impl Solution {
    pub fn good_subsetof_binary_matrix(grid: Vec<Vec<i32>>) -> Vec<i32> {
        let mut ans: Vec<i32> = vec![];
        let mut mp:HashMap<i32, i32> = HashMap::new();
        let m = grid.len();
        let n = grid[0].len();
        for j in 0..m {
            let mut st = 0;
            for i in 0..n {
                st |= grid[j][i] << i;
            }
            mp.insert(st, j as i32);
        }
        if mp.contains_key(&0) {
            ans.push(*mp.get(&0).unwrap());
            return ans;
        }
        for (&x, &i) in &mp {
            for (&y, &j) in &mp {
                if x & y == 0 {
                    return vec![i.min(j), i.max(j)];
                }
            }
        }
        ans
    }
}

fn main() {
    println!("{:#?}", Solution::good_subsetof_binary_matrix(vec![
        vec![0,1,1,0],
        vec![0,0,0,1],
        vec![1,1,1,1],
    ]))
}